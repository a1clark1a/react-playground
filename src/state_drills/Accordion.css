.Accordion {
  list-style-type: none;
  padding: 0;
}

button {
  height: 50px;
  width: 100px;
  margin: 20px 5px 20px 5px;
}

button:hover {
  background-color: gray;
  cursor: pointer;
}

li > p {
  background-color: teal;
  padding: 10px 0 10px 0;
  color: wheat;
}
