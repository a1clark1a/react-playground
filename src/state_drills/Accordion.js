import React from "react";
import "./Accordion.css";

//Accordion component accepts a prop named sections
//sections prop is an array of objects
class Accordion extends React.Component {
  static defaultProps = {
    sections: []
  };

  state = {
    currentSectionIndex: null
  };

  handleClick(index) {
    this.setState({
      currentSectionIndex: index
    });
  }

  renderList() {
    const { sections } = this.props;
    const { currentSectionIndex } = this.state;
    return sections.map((objSect, i) => {
      return (
        <li key={i}>
          <button type="button" key={i} onClick={() => this.handleClick(i)}>
            {objSect.title}
          </button>
          {currentSectionIndex === i && <p>{objSect.content}</p>}
        </li>
      );
    });
  }

  render() {
    return <ul className="Accordion">{this.renderList()}</ul>;
  }
}

export default Accordion;
