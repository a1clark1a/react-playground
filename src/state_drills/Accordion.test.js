import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Accordion from "./Accordion";

describe("Accordion Component", () => {
  const sections = [
    {
      title: "Section 1",
      content: "Lorem ipsum dolor sit amet consectetur adipisicing elit."
    },
    {
      title: "Section 2",
      content:
        "Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!"
    },
    {
      title: "Section 3",
      content:
        "Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?"
    }
  ];

  it("renders without errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Accordion sections={sections} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders no content by default", () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders empty if no sections give", () => {
    const wrapper = shallow(<Accordion />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("content opens when any section is clicked", () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    wrapper
      .find("button")
      .at(1)
      .simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Only one content from one section opens at a time", () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    wrapper.find("button").forEach(button => {
      return button.simulate("click");
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
