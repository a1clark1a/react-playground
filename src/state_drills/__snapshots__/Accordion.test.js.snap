// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component Only one content from one section opens at a time 1`] = `
<ul
  className="Accordion"
>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion Component content opens when any section is clicked 1`] = `
<ul
  className="Accordion"
>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty if no sections give 1`] = `
<ul
  className="Accordion"
/>
`;

exports[`Accordion Component renders no content by default 1`] = `
<ul
  className="Accordion"
>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;
